pkg/tidydns/tidydns.go

```diff
@@ -84,15 +112,248 @@ const (
 	RecordTypeSSHFP RecordType = 8
 	RecordTypeTLSA  RecordType = 9
 	RecordTypeCAA   RecordType = 10
+
+	AuthGroupUser       AuthGroup = 2
+	AuthGroupSuperAdmin AuthGroup = 1
 )
 
+const errorTidyDNS = "error from tidyDNS server: %s"
+const headerContentType = "Content-Type"
+const mimeForm = "application/x-www-form-urlencoded"
+
 type tidyDNSClient struct {
 	client   *http.Client
 	username string
 	password string
 	baseURL  string
 }
 
+func (c *tidyDNSClient) CreateInternalUser(ctx context.Context, username string, password string, description string, changePasswordOnFirstLogin bool, authGroup AuthGroup, userAllow []UserAllowID) (UserID, error) {
+	var userAllowFormatted []string
+	if len(userAllow) > 0 {
+		userAllowFormatted = make([]string, 0, len(userAllowFormatted))
+		for _, id := range userAllow {
+			userAllowFormatted = append(userAllowFormatted, strconv.Itoa(int(id)))
+		}
+	} else {
+		userAllowFormatted = []string{""}
+	}
+
+	var changePasswordOnFirstLoginFormatted string
+	if changePasswordOnFirstLogin {
+		changePasswordOnFirstLoginFormatted = "1"
+	} else {
+		changePasswordOnFirstLoginFormatted = "0"
+	}
+
+	data := url.Values{
+		"username":                       {username},
+		"epassword":                      {password},
+		"epassword_verify":               {password},
+		"change_password_on_first_login": {changePasswordOnFirstLoginFormatted},
+		"description":                    {description},
+		"auth_group":                     {strconv.Itoa(int(authGroup))},
+		//"tmp_auth_group":                 {""},
+		"user_allow": userAllowFormatted,
+	}
+
+	req, err := http.NewRequestWithContext(ctx, "POST", fmt.Sprintf("%s/=/user/new", c.baseURL), strings.NewReader(data.Encode()))
+	if err != nil {
+		return 0, err
+	}
+	req.SetBasicAuth(c.username, c.password)
+	req.Header.Set(headerContentType, mimeForm)
+
+	res, err := c.client.Do(req)
+	if err != nil {
+		return 0, err
+	}
+	defer func(Body io.ReadCloser) {
+		_ = Body.Close()
+	}(res.Body)
+	if res.StatusCode != http.StatusOK {
+		bodyBytes, err := io.ReadAll(res.Body)
+		bodyString := string(bodyBytes)
+		if err != nil {
+			return 0, err
+		}
+		if strings.Contains(bodyString, fmt.Sprintf("Key (username)=(%s) already exists", username)) {
+			return 0, fmt.Errorf("username already exists")
+		}
+		return 0, fmt.Errorf(errorTidyDNS, res.Status)
+	}
+
+	var user userCreate
+	err = json.NewDecoder(res.Body).Decode(&user)
+	if err != nil {
+		return 0, err
+	}
+
+	return UserID(user.Data.Id), nil
+}
+
+func (c *tidyDNSClient) GetInternalUser(ctx context.Context, userID UserID) (*UserInfo, error) {
+	req, err := http.NewRequestWithContext(
+		ctx,
+		"GET",
+		fmt.Sprintf("%s/=/user/%s", c.baseURL, strconv.Itoa(int(userID))),
+		nil,
+	)
```

Comment:
In these situations it could be formatted like:
```go
url := fmt.Sprintf("%s/=/user/%d", c.baseURL, userID)
req, err := http.NewRequestWithContext(ctx, http.MethodGet, url, nil)
```

Reply:
refactored to match this format in 8af455f958157e535c862e83bfb021205b0e0e32